packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -273,22 +276,28 @@ function WorklistDataGrid(props) {
     if (hasSelectableRows && focusedRow === 0 && focusedCol === 0) {
       // When row selection is turned on for the first time, default to the first row selection cell.
       newFocusCell = { row: 1, col: 0 };
-    } else if (!hasSelectableRows && focusedCol === 0) {
+    } else if (!hasSelectableRows) {
       // When row selection is turned off, if a cell in the row selection had focus, then
       // refocus on the first cell in that row.
-      newFocusCell = { row: focusedRow, col: 0 };
+      if (focusedCol === 0) {
+        newFocusCell = { row: focusedRow, col: 0 };
+      }
+
+      selectedRows.current = [];
     }
     setFocusedRowCol(newFocusCell.row, newFocusCell.col, false);
 
     if (!hasSelectableRows) {
       multiSelectRange.current = {};
     }
 
-    // Since the row selection mode has changed, the row selection mode needs to be updated.
-    if (!isRowSelectionModeToggledByGrid.current) {
-      setAriaLiveMessage(intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
+    if (!rowSelectionEffectTriggered.current) {
+      rowSelectionEffectTriggered.current = true;
+      return;
     }
-    isRowSelectionModeToggledByGrid.current = false;
+
+    // Since the row selection mode has changed, the row selection mode needs to be updated.
+    setRowSelectionModeAriaLiveMessage(intl.formatMessage({ id: hasSelectableRows ? 'Terra.worklist-data-grid.row-selection-mode-enabled' : 'Terra.worklist-data-grid.row-selection-mode-disabled' }));
```

Comment:
nit: 
```suggestion
    setRowSelectionModeAriaLiveMessage(intl.formatMessage({ id: '`Terra.worklist-data-grid.row-selection-mode-${hasSelectableRows ? "enabled" : "disabled"}` }));
```

This is a pretty long line, I think we could simplify it like this!